### PR TITLE
Improve consistency, updates dependencies (Kubernetes 1.30), simplify workflow/stack names

### PR DESCRIPTION
**Pull request by NickAnderegg**
This PR does a few separate things that are best reviewed as a package, but which were added as atomic commits for easier review (and reversion if necessary) of each change. The changes are the following:

- https://github.com/workflows-sh/aws-eks-ec2-asg-cdk/commit/aed58f2860cd8dfea22f42e7b4c322d3866bc805: REVERTED - But kept changes that simplified name of sample app repo to reflect the changes implemented in https://github.com/workflows-sh/sample-expressjs-aws-eks-ec2-asg-cdk/pull/185

  ~~**Simplify the names of the workflows and stacks for improved easy of use and readability.**~~

  ~~Instances of the string `aws-eks-ec2-asg-cdk` in workflow/stack names and `AWS_EKS_EC2_ASG_CDK` in environment variables were replaced with `aws-eks-stack` and `AWS_EKS_STACK`. A string of mostly-arbitrary letters is much more difficult to parse, so the other used resources beyond EKS are simply amalgamated into "stack". This will allow us to differentiate between the stack based on EKS and the stack based on Fargate, if we are continuing to keep that one at feature parity. To go along with this change, I also incremented the workflows' version to `0.2.0`~~

- https://github.com/workflows-sh/aws-eks-ec2-asg-cdk/commit/776acc0085bcd017e0a0855a4eb8f75f25b93352: Updates cluster definition to use Kubernetes 1.30, and updates the related dependencies accordingly.
- https://github.com/workflows-sh/aws-eks-ec2-asg-cdk/commit/3646322cbf84e43b5f4f8adb3d36684c34fd9e3a: Minor improvement to the workflow prompts to have them use `list` prompts, rather than `input` prompts, as we constrain the possible input values to {dev, stg, prd, all}.
- https://github.com/workflows-sh/aws-eks-ec2-asg-cdk/commit/7b80e497133e3b27ffb72b7cc49fc769fd93a102: Replaces the deprecated method being used to define a subnet for the VPC.
- https://github.com/workflows-sh/aws-eks-ec2-asg-cdk/commit/90fd24f79716b5817bc63b80253e545cc7aa4cc1: Explicitly associates the cluster's master role to the cluster by the cluster's name, which resolves some odd permission issues I was encountering.
- https://github.com/workflows-sh/aws-eks-ec2-asg-cdk/commit/017f4bf1741f375fc24e9621e1f81f6ddaf484cb: In the src/stack/index.ts file where the individual stacks for our environments are defined, manually add dependencies between the stacks to ensure everything is torn down in the correct order when the stack is destroyed.

### CODE REVIEW

**Review by nitishkumar71 — APPROVED** [review]
LGTM. If we are ok with name change.

**Review by jossemargt-cto-ai — COMMENTED** [review]
Thanks for all the stack upgrades you did here. 

Still, let's get the approval for the stack (and derivated resources) name change with the solutions team before proceeding with the merge.

**Review by jossemargt-cto-ai — APPROVED** [review]

**Review by jossemargt-cto-ai — APPROVED** [review]
Since we are no longer renaming the stack's name LGTM

**Review by nitishkumar71 — APPROVED** [review]

**Pull request merged**
